src/globals.c

```diff
@@ -688,3 +688,19 @@ int global_parse_options(int argc, char **argv, const struct global_options *opt
 	free(opts_array);
 	return ret ? ret : optind;
 }
+
+bool global_parse_csv(const char *args, struct list **parsed_args)
+{
+	char *arg_copy = strdup_or_die(args);
+	char *token = strtok(arg_copy, ",");
+	while (token) {
+		*parsed_args = list_prepend_data(*parsed_args, strdup_or_die(token));
+		token = strtok(NULL, ",");
+	}
+	free(arg_copy);
+	if (!*parsed_args) {
```

Comment:
if global_parse_args() is always returning if *parsed_args is NULL or not we can just return that value. So we can have:

    struct list *global_parse_csv(const char *args);
    ...
    my_list = global_parse_csv(my_string);
    if (my_list) {
         //it's ok
    } else {
        print("error");
    }

And then you can even move that function to string.h/string.c so it can be reused by other code. string_parse_csv(), or something like that.

Reply:
it's the inversion of string_join(). To make it more generic you can even get the "," as argument too